// react-ui-kanban/dist/dnd/Container.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _reactDom = _interopRequireDefault(require("react-dom"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _trelloSmoothDnd = _interopRequireWildcard(require("trello-smooth-dnd"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
_trelloSmoothDnd.default.dropHandler = _trelloSmoothDnd.dropHandlers.reactDropHandler().handler;
_trelloSmoothDnd.default.wrapChild = p => p; // dont wrap children they will already be wrapped

class Container extends _react.Component {
  constructor(props) {
    super(props);
    this.getContainerOptions = this.getContainerOptions.bind(this);
    this.setRef = this.setRef.bind(this);
    this.prevContainer = null;
  }
  componentDidMount() {
    this.containerDiv = this.containerDiv || _reactDom.default.findDOMNode(this);
    this.prevContainer = this.containerDiv;
    this.container = (0, _trelloSmoothDnd.default)(this.containerDiv, this.getContainerOptions());
  }
  componentWillUnmount() {
    this.container.dispose();
    this.container = null;
  }
  componentDidUpdate() {
    this.containerDiv = this.containerDiv || _reactDom.default.findDOMNode(this);
    if (this.containerDiv) {
      if (this.prevContainer && this.prevContainer !== this.containerDiv) {
        this.container.dispose();
        this.container = (0, _trelloSmoothDnd.default)(this.containerDiv, this.getContainerOptions());
        this.prevContainer = this.containerDiv;
      }
    }
  }
  render() {
    if (this.props.render) {
      return this.props.render(this.setRef);
    } else {
      return /*#__PURE__*/_react.default.createElement("div", {
        style: this.props.style,
        ref: this.setRef
      }, this.props.children);
    }
  }
  setRef(element) {
    this.containerDiv = element;
  }
  getContainerOptions() {
    var _this = this;
    var functionProps = {};
    if (this.props.onDragStart) {
      functionProps.onDragStart = function () {
        return _this.props.onDragStart(...arguments);
      };
    }
    if (this.props.onDragEnd) {
      functionProps.onDragEnd = function () {
        return _this.props.onDragEnd(...arguments);
      };
    }
    if (this.props.onDrop) {
      functionProps.onDrop = function () {
        return _this.props.onDrop(...arguments);
      };
    }
    if (this.props.getChildPayload) {
      functionProps.getChildPayload = function () {
        return _this.props.getChildPayload(...arguments);
      };
    }
    if (this.props.shouldAnimateDrop) {
      functionProps.shouldAnimateDrop = function () {
        return _this.props.shouldAnimateDrop(...arguments);
      };
    }
    if (this.props.shouldAcceptDrop) {
      functionProps.shouldAcceptDrop = function () {
        return _this.props.shouldAcceptDrop(...arguments);
      };
    }
    if (this.props.onDragEnter) {
      functionProps.onDragEnter = function () {
        return _this.props.onDragEnter(...arguments);
      };
    }
    if (this.props.onDragLeave) {
      functionProps.onDragLeave = function () {
        return _this.props.onDragLeave(...arguments);
      };
    }
    if (this.props.render) {
      functionProps.render = function () {
        return _this.props.render(...arguments);
      };
    }
    if (this.props.onDropReady) {
      functionProps.onDropReady = function () {
        return _this.props.onDropReady(...arguments);
      };
    }
    if (this.props.getGhostParent) {
      functionProps.getGhostParent = function () {
        return _this.props.getGhostParent(...arguments);
      };
    }
    return Object.assign({}, this.props, functionProps);
  }
}
Container.propTypes = {
  behaviour: _propTypes.default.oneOf(["move", "copy", "drag-zone"]),
  groupName: _propTypes.default.string,
  orientation: _propTypes.default.oneOf(["horizontal", "vertical"]),
  style: _propTypes.default.object,
  dragHandleSelector: _propTypes.default.string,
  className: _propTypes.default.string,
  nonDragAreaSelector: _propTypes.default.string,
  dragBeginDelay: _propTypes.default.number,
  animationDuration: _propTypes.default.number,
  autoScrollEnabled: _propTypes.default.string,
  lockAxis: _propTypes.default.string,
  dragClass: _propTypes.default.string,
  dropClass: _propTypes.default.string,
  onDragStart: _propTypes.default.func,
  onDragEnd: _propTypes.default.func,
  onDrop: _propTypes.default.func,
  getChildPayload: _propTypes.default.func,
  shouldAnimateDrop: _propTypes.default.func,
  shouldAcceptDrop: _propTypes.default.func,
  onDragEnter: _propTypes.default.func,
  onDragLeave: _propTypes.default.func,
  render: _propTypes.default.func,
  getGhostParent: _propTypes.default.func,
  removeOnDropOut: _propTypes.default.bool
};
Container.defaultProps = {
  behaviour: 'move',
  orientation: 'vertical',
  className: 'reactTrelloBoard'
};
var _default = exports.default = Container;
